LanguageFeatures/Views/syntax_A04_t01.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A rule for <viewDeclaration> is added to the grammar, along with
+/// some rules for elements used in view declarations:
+///
+/// <viewDeclaration> ::=
+///   'view' 'class' <typeIdentifier> <typeParameters>?
+///       <viewPrimaryConstructor>?
+///       <interfaces>?
+///   '{'
+///     (<metadata> <viewMemberDeclaration>)*
+///   '}'
+///
+/// <viewPrimaryConstructor> ::=
+///   '(' <type> <identifier> ')'
+///
+/// <viewMemberDeclaration> ::=
+///   <classMemberDefinition>
+/// ...
+/// The name of the representation in a view declaration that includes a
+/// <viewPrimaryConstructor> is the identifier id specified in there, and the
+/// type of the representation is the declared type of id.
+///
+/// @description Checks that the name of the representation in a view
+/// declaration that includes a <viewPrimaryConstructor> is the identifier id
+/// specified in there, and the type of the representation is the declared type
+/// of id
```

Comment:
Can we actually test that? What does it mean? (This sentence serves to define the phrases 'the name of the representation' and 'the type of the representation'.)

The name and type of the representation is just the name and type of `id0` and `_x` in the examples below. So we should be testing statically that it behaves like a normal instance variable of a class: The type of `v1.id0` is num, and it has a getter and no setter. Similarly for `_x`.

Dynamically we would test the things being tested below (to make sure the value is stored (and not just ignored and replaced by null or whatever), but also that the "view object" does not exist at run time, that is, that `v1 is num`.

There may be static tests below, otherwise we should have them.